pkg/pillar/devicenetwork/dhcpcd.go

```diff
@@ -19,29 +17,41 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
 )
 
 // UpdateDhcpClient starts/modifies/deletes dhcpcd per interface
 // Returns an ifname and error if an interface does not yet exist
-func UpdateDhcpClient(newConfig, oldConfig types.DevicePortConfig) (string, error) {
+func UpdateDhcpClient(newConfig, oldConfig types.DevicePortConfig) types.IntfStatusMap {
 
 	// Look for adds or changes
 	log.Infof("updateDhcpClient: new %v old %v\n",
 		newConfig, oldConfig)
 	// Dry-run to see if we need to ask for retry. Don't change anything
```

Comment:
This comment is key. When returning a failure you can't change any of the DHCP configuration.
If you do the caller will be confused and will fail to do the correct doDhcpClientActivate/doDhcpClientInactivate next time around; the dryrun aspect is critical. Please read the history of this code before changing it.